src/node_crypto.cc

```diff
@@ -4803,8 +4803,8 @@ void PBKDF2(const FunctionCallbackInfo<Value>& args) {
     goto err;
   }
 
-  iter = args[2]->Int32Value();
-  if (iter < 0) {
+  iter = args[2]->NumberValue();
+  if (iter < 0 || isnan(iter) || isinf(iter)) {
     type_error = "Bad iterations";
```

Comment:
would bump this to a semver-major. let's do that in either a difference commit or PR so it can be backported.